py-polars/polars/utils/udfs.py

```diff
@@ -178,6 +182,35 @@ def _inject_nesting(
 
         return sorted(expression_blocks.items())
 
+    def _get_target_name(self, col: str, expression: str) -> str:
+        """The name of the object against which the 'apply' is being invoked."""
+        if self._apply_target_name is not None:
+            return self._apply_target_name
+        else:
+            col_expr = f'pl.col("{col}")'
+            if self._apply_target == "expr":
+                return col_expr
+            elif self._apply_target == "series":
+                # handle overlapping name from global variables; fallback through
+                # "s", "srs", "series", and then the series' name (if applicable)
+                series_aliases = ["s", "srs", "series"]
+                if col.isalpha() and col not in series_aliases:
+                    series_aliases.append(col)
+                search_expr = expression.replace(col_expr, "")
+                for name in series_aliases:
+                    if not re.search(rf"\b{name}\b", search_expr):
+                        self._apply_target_name = name
+                        return name
+
+                # ...before, finally, srs0 -> srsN until we find an available name
+                n = count()
+                while True:
+                    name = f"srs{next(n)}"
+                    if not re.search(rf"\bs{name}\b", search_expr):
```

Comment:
why is there an extra 's' here? why not just 
```suggestion
                    if not re.search(rf"\b{name}\b", search_expr):
```
?

Reply:
Typo :)